presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleAssert.java

```diff
@@ -91,33 +92,29 @@ public RuleAssert on(Function<PlanBuilder, PlanNode> planProvider)
 
     public void doesNotFire()
     {
-        SymbolAllocator symbolAllocator = new SymbolAllocator(symbols);
-        Optional<PlanNode> result = inTransaction(session -> rule.apply(plan, x -> x, idAllocator, symbolAllocator, session));
+        RuleApplication ruleApplication = applyRule();
 
-        if (result.isPresent()) {
+        if (ruleApplication.result.isPresent()) {
             fail(String.format(
                     "Expected %s to not fire for:\n%s",
                     rule.getClass().getName(),
-                    inTransaction(session -> PlanPrinter.textLogicalPlan(plan, symbolAllocator.getTypes(), metadata, costCalculator, session, 2))));
+                    inTransaction(session -> PlanPrinter.textLogicalPlan(plan, ruleApplication.types, metadata, costCalculator, session, 2))));
```

Comment:
you can also extract this long line as method so you would not change it in the other place